Adding GravitySensor API documentation.

arskama opened the pull request:
Fixes #3524
Implementation of GravitySensor API for Chrome 91 [1]
Dependency with update of compat-data [2]
[1] https://chromestatus.com/features/5384099747332096
[2] https://github.com/mdn/browser-compat-data/pull/9628

arskama committed:
Adding GravitySensor API documentation.

Fixes #3524
Implementation of GravitySensor API for Chrome 91 [1]
Dependency with update of compat-data [2]
[1] https://chromestatus.com/features/5384099747332096
[2] https://github.com/mdn/browser-compat-data/pull/9628

wbamberg commented on files/en-us/web/api/gravitysensor/gravitysensor/index.html:
```diff
@@ -0,0 +1,73 @@
+    constructor creates a new {{domxref("GravitySensor")}} object which
+    provides on each reading the gravity applied to the device along all three axes.</span></p>
+
+<p>If a feature policy blocks use of a feature, it is because your code is inconsistent
```
It feels inappropriate to have this paragraph right at the start like this. Is this really the second most important thing someone needs to know about the API?

If "If a feature policy blocks use of a feature", how would this be manifested? Would an exception be thrown in the constructor? If so, I would expect this would be documented in an "Exceptions" heading, like https://developer.mozilla.org/en-US/docs/Web/API/WritableStreamDefaultWriter/WritableStreamDefaultWriter#exceptions.

arskama replied:
comment tackled in:
https://github.com/mdn/content/commit/7e7651d998db5ff6156dd6185085bde6f5aeeac7

wbamberg commented on files/en-us/web/api/gravitysensor/gravitysensor/index.html:
```diff
@@ -0,0 +1,73 @@
+        {{domxref('sensor.onreading')}} will be called. A whole number or decimal may be
+        used, the latter for frequencies less than a second. The actual reading frequency
+        depends on device hardware and consequently may be less than requested.</li>
+      <li><code>referenceFrame</code>: Either <code>'device'</code> or
```
We should explain what this option means.

arskama replied:
comment tackled in:
https://github.com/mdn/content/commit/7e7651d998db5ff6156dd6185085bde6f5aeeac7

Not sure if needed more information, but otherwise a link to the specs is needed maybe.

wbamberg replied:
I think what you have here is fine. We have a link to the spec in "Specifications" and people can refer to that if the need to.

wbamberg commented on files/en-us/web/api/gravitysensor/gravitysensor/index.html:
```diff
@@ -0,0 +1,73 @@
+  <dt><em>options</em> {{optional_inline}}</dt>
+  <dd>Options are as follows:
+    <ul>
+      <li><code>frequency</code>: The desired number of times per second a sample should
```
Since this is optional we should describe the default.

arskama replied:
done in
https://github.com/mdn/content/commit/7e7651d998db5ff6156dd6185085bde6f5aeeac7

wbamberg commented on files/en-us/web/api/gravitysensor/index.html:
```diff
@@ -0,0 +1,72 @@
+
+<p>To use this sensor, the user must grant permission to the <code>'accelerometer'</code> device sensor through the {{domxref('Permissions')}} API.</p>
+
+<p>If a feature policy blocks use of a feature it is because your code is inconsistent with the policies set on your server. This is not something that would ever be shown to a user. See {{httpheader('Feature-Policy')}} for implementation instructions.</p>
```
Same comment as above about this paragraph.

arskama replied:
removed. I guess, if explained on the constructor, that is enough.

wbamberg commented on files/en-us/web/api/gravitysensor/index.html:
```diff
@@ -0,0 +1,72 @@
+
+<p>Gravity is typically read in the {{domxref('Sensor.onreading')}} event callback. In the example below this occurs sixty times a second.</p>
+
+<pre class="brush: js">let gravitySensor = new GravitySensor({frequency: 60});
+
+gravitySensor.addEventListener('reading', e =&gt; {
+  console.log("Gravity along the X-axis " + gravitySensor.x);
+  console.log("Gravity along the Y-axis " + gravitySensor.y);
+  console.log("Gravity along the Z-axis " + gravitySensor.z);
+});
+laSensor.start();</pre>
```
```suggestion
<pre class="brush: js">const gravitySensor = new GravitySensor({frequency: 60});

gravitySensor.addEventListener("reading", e =&gt; {
  console.log(`Gravity along the X-axis ${gravitySensor.x}`);
  console.log(`Gravity along the Y-axis ${gravitySensor.y}`);
  console.log(`Gravity along the Z-axis ${gravitySensor.z}`);
});

gravitySensor.start();</pre>
```

arskama replied:
done

wbamberg commented on files/en-us/web/api/sensor_apis/index.html:
```diff
@@ -175,6 +179,8 @@ <h2 id="interfaces">Interfaces</h2>
  <dt>{{domxref('AmbientLightSensor')}}{{securecontext_inline}}</dt>
  <dd>Returns the current light level or illuminance of the ambient light around the hosting device.</dd>
+ <dt>{{domxref('GravitySensor')}}{{securecontext_inline}}</dt>
+ <dd>Provides the gravity applied to the device along all three axes, but without the contribution of human artifact.</dd>
```
I don't understand what "without the contribution of human artifact" means here.

arskama replied:
removed, it was confusing.

wbamberg commented on files/en-us/web/api/sensor_apis/index.html:
```diff
@@ -208,7 +214,7 @@ <h2 id="Specifications">Specifications</h2>
    <td>{{SpecName('Accelerometer')}}</td>
    <td>{{Spec2('Accelerometer')}}</td>
-   <td>Defines <code>Accelerometer</code> and <code>LinearAccerationSensor</code>.</td>
+   <td>Defines <code>Accelerometer</code>, <code>GravitySensor</code> and <code>LinearAccelerationSensor</code>.</td>
```
```suggestion
   <td>Defines <code>Accelerometer</code>, <code>GravitySensor</code>, and <code>LinearAccelerationSensor</code>.</td>
```

arskama replied:
done

wbamberg reviewed (changes requested):
Thanks for your contribution to MDN, @arskama ! I had a few comments.

arskama committed:
Fixing comments GravitySensor API documentation.

Fixes #3524
Implementation of GravitySensor API for Chrome 91 [1]
Dependency with update of compat-data [2]
[1] https://chromestatus.com/features/5384099747332096
[2] https://github.com/mdn/browser-compat-data/pull/9628

arskama commented:
@wbamberg. Thanks for the review!

wbamberg commented on files/en-us/web/api/gravitysensor/gravitysensor/index.html:
```diff
@@ -0,0 +1,80 @@
+        used, the latter for frequencies less than a second. The actual reading frequency
+        depends on device hardware and consequently may be less than requested. The default
+        frequency is the one defined by the underlying platform</li>
+      <li><code>referenceFrame</code>: The local coordinate system represents 
```
```suggestion
      <li><code>referenceFrame</code>: The local coordinate system representing 
```

arskama replied:
fixed

wbamberg commented on files/en-us/web/api/gravitysensor/gravitysensor/index.html:
```diff
@@ -0,0 +1,80 @@
+        {{domxref('sensor.onreading')}} will be called. A whole number or decimal may be
+        used, the latter for frequencies less than a second. The actual reading frequency
+        depends on device hardware and consequently may be less than requested. The default
+        frequency is the one defined by the underlying platform</li>
```
```suggestion
        frequency is the one defined by the underlying platform.</li>
```

arskama replied:
fixed

wbamberg commented on files/en-us/web/api/gravitysensor/gravitysensor/index.html:
```diff
@@ -0,0 +1,80 @@
+
+<dl>
+  <dt>SecurityError</dt>
+  <dd>If a feature policy blocks use of a feature, it is because your code is inconsistent
```
```suggestion
  <dd>Use of this feature was blocked by a feature policy. If a feature policy blocks use of a feature, it is because your code is inconsistent
```

It would be good to have an introductory sentence, like the one in this suggestion, to link this exception to the feature policy bit.

arskama replied:
correct.

wbamberg reviewed (changes requested):
Thanks for the updates, @arskama ! I saw a couple of copy-edits, and think we ought to have a sentence in the Exceptions bit, as suggested.

arskama committed:
Fixing comments part 2 GravitySensor API documentation.

Fixes #3524
Implementation of GravitySensor API for Chrome 91 [1]
Dependency with update of compat-data [2]
[1] https://chromestatus.com/features/5384099747332096
[2] mdn/browser-compat-data#9628

arskama reviewed (commented):
review comments fixed in 8c010d1d98ccde0f2eddd8e300f0cb09ed7c3a67

wbamberg commented on files/en-us/web/api/gravitysensor/gravitysensor/index.html:
```diff
@@ -0,0 +1,80 @@
+
+<dl>
+  <dt>SecurityError</dt>
+  <dd>Use of this feature was blocked by feature policy. If a feature policy blocks use of a feature,
```
```suggestion
  <dd>Use of this feature was blocked by a feature policy. If a feature policy blocks use of a feature,
```

wbamberg committed:
Update files/en-us/web/api/gravitysensor/gravitysensor/index.html

wbamberg reviewed (approved):
Thanks @arskama , looks great!

arskama commented:
Thanks @wbamberg for merging the patch.

I can see the pages now, but I was wondering why i don't see the GravitySensor in the list. (locally I do see it)
https://developer.mozilla.org/en-US/docs/Web/API/Sensor#interfaces_based_on_sensor
Am I missing something?

I was also wondering why a similar page, such as:
https://developer.mozilla.org/en-US/docs/Web/API/LinearAccelerationSensor/LinearAccelerationSensor
has a Inheritance sub-section on the Related Topics section but 
https://developer.mozilla.org/en-US/docs/Web/API/GravitySensor doesn't have that sub-section.

Thanks in advance.
